kasa/aestransport.py

```diff
@@ -223,7 +225,8 @@ async def try_login(self, login_params: Dict[str, Any]) -> None:
 
         resp_dict = await self.send_secure_passthrough(request)
         self._handle_response_error_code(resp_dict, "Error logging in")
-        self._login_token = resp_dict["result"]["token"]
+        login_token = resp_dict["result"]["token"]
+        self._token_url = URL(f"{self._app_url}?token={login_token}")
```

Comment:
If this was properly passed to the aiohtttp library as a parameter argument instead of appending it to the url string would it still be expensive?

Reply:
If you pass a `URL` object to `aiohttp` it doesn't have to build it.

Reply:
Just realized it would be a bit cleaner to have written it as `.with_query` so I made that adjustment

Reply:
I just mean pass the token as the `params` argument but still pass the base url as `URL`. Like the Klap transport does with seq. That’s really how it should be done anyway.

Reply:
Yes so I just tested this out and the following change means you don't need to change the token logic to have it's own separate URL:

```
-        url = f"http://{self._host}/app"
-        if self._state is TransportState.ESTABLISHED and self._login_token:
-            url += f"?token={self._login_token}"
+        url = self._app_url  # The URL object created earlier
+        params = {"token", self._login_token} if self._state is TransportState.ESTABLISHED and self._login_token else None
```
Then just pass the params and aiohttp doesn't create a new URL object:
```
        status_code, resp_dict = await self._http_client.post(
            url,
            json=passthrough_request,
             headers=self.COMMON_HEADERS,
             cookies_dict=self._session_cookie,
+            params=params,
         )
```
I wouldn't normally have added the parameters directly onto the url in the first place but I lifted that line of code and forgot to change it to pass `param` as an arg.

Reply:
Will do some tracing with that change later today. 

Will need to check if it makes a new URL object in that case (which we want to avoid)

Reply:
Its going to call `url.update_query(params)` so we end up with a new URL object which we want to avoid.  Let's keep it as-is

Reply:
It looks like making a new URL isn't as bad with `update_query` because It makes a new URL <https://github.com/aio-libs/yarl/blob/d472b94c5f969c9758f5a0107a57d5a3e8090aa8/yarl/_url.py#L1011> has `encoded=False` which avoids the https://github.com/aio-libs/yarl/blob/d472b94c5f969c9758f5a0107a57d5a3e8090aa8/yarl/_url.py#L791 slow path

However its still making a new URL object each time in this case so still think its best to leave as-is.